src/crypto-providers/signingValidation.ts

```diff
@@ -184,21 +210,26 @@ const validateWitnessing = (params: SigningParameters): void => {
 }
 
 const validateSigning = (params: SigningParameters): void => {
-  if (params.rawTx.body.inputs.length === 0) {
-    throw Error(Errors.MissingInputError)
-  }
+  const body = (params.rawTx?.body ?? params.tx?.body)!
+  const { hwSigningFileData } = params
+
+  validateNetworkId(params.network.networkId, body.networkId)
 
   switch (params.signingMode) {
     case SigningMode.ORDINARY_TRANSACTION:
-      validateOrdinarySigning(params)
+      validateOrdinarySigning(body, hwSigningFileData)
       break
 
     case SigningMode.POOL_REGISTRATION_AS_OWNER:
     case SigningMode.POOL_REGISTRATION_AS_OPERATOR:
       throw Error(Errors.CantSignTxWithPoolRegError)
 
     case SigningMode.MULTISIG_TRANSACTION:
-      validateMultisigSigning(params)
+      validateMultisigSigning(body, hwSigningFileData)
+      break
+
+    case SigningMode.PLUTUS_TRANSACTION:
+      validatePlutusSigning(body, hwSigningFileData)
```

Comment:
This specifically (PLUTUS_TRANSACTION with sigining call) doesn't need to be supported, be forbidding it would bring no benefit I think. What would maybe help would be dropping support for Plutus-era txs with signing call, because that would simplify the signed tx assembling code (we decided we want to deprecate signed tx assembling only after this was implemented). But at the moment, I see no reason to remove it now - we will remove the entire signed tx assembling in a few months, and in case there are problems with Plutus-era txs with signing call, we can still point users to the witness call instead of forbidding it right now.